packages/strapi-design-system/src/Select/Option.tsx

```diff
@@ -113,14 +72,62 @@ Option.defaultProps = {
   indeterminate: false,
 };
 
-Option.propTypes = {
-  children: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
-  indeterminate: PropTypes.bool,
-  isChild: PropTypes.bool,
-  multi: PropTypes.bool,
-  selected: PropTypes.bool,
-  startIcon: PropTypes.node,
-  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
-};
-
 Option.displayName = 'Option';
+
+interface CheckMarkProps {
+  selected?: boolean;
+  indeterminate?: boolean;
+}
+
+const CheckMark = styled(Box)<CheckMarkProps>`
+  border: 1px solid
+    ${({ theme, selected, indeterminate }) =>
+      selected || indeterminate ? theme.colors.primary600 : theme.colors.neutral300};
+  height: 18px;
+  width: 18px;
+  background-color: ${({ theme, selected, indeterminate }) =>
+    selected || indeterminate ? theme.colors.primary600 : theme.colors.neutral0};
+
+  ${({ theme, indeterminate }) =>
+    indeterminate &&
+    css`
+      &::after {
+        content: '';
+        display: block;
+        position: relative;
+        color: white;
+        height: 2px;
+        width: 10px;
+        background-color: ${theme.colors.neutral0};
+        left: 50%;
+        top: 50%;
+        transform: translateX(-50%) translateY(-50%);
+      }
+    `}
+
+  ${({ selected }) =>
+    selected &&
+    css`
+      &::after {
+        content: '';
+        background: url(${checkmarkIcon}) no-repeat no-repeat center center;
+        width: 100%;
+        height: 100%;
+        position: absolute;
+      }
+    `}
+`;
+
+const OptionBox = styled(Box)`
+  width: 100%;
```

Comment:
Same for this one: could be a prop instead.

Reply:
This will be gone in the combobox refactor